lsp/nls/tests/snapshots/main__lsp__nls__tests__inputs__completion-dict.ncl.snap

```diff
@@ -2,6 +2,6 @@
 source: lsp/nls/tests/main.rs
 expression: output
 ---
-[foo]
-[foo]
+[foo, x]
+[Dict, foo, x]
```

Comment:
Same here, why we have completion for both variable from the environment and dictionary fields? It seems like we are conflating completion from the current environment (variables in scope) and field completion (triggered by a `.`). But those are arguably different, and might be a regression: when I write `myrecord.`, variables in scope have no reason to be valid input here - we should only suggest completion from what fields can be inferred from `myrecord`. Same for context completion. On the other hand, non-record completion should only show things in scope.

Reply:
I think I agree with you (it's what I intended to bring up in the last paragraph of the PR description). The previous nls behavior was using both completion sources when completing a record path, so I was just copying that behavior.

I'll make the change to turn of the variables-in-scope completion when completing record fields, and we can see if we like the results.

Reply:
Ah, I see, I think I didn't really understand that paragraph, and once I had sufficient context could I had just forget about it :sweat_smile: